src/polymer/polymer2-element-scanner.ts

```diff
@@ -65,16 +65,25 @@ class ElementVisitor implements Visitor {
     }
     const element = this._handleClass(node);
     if (element) {
-      element.className = className;
-      this._possibleElements.set(element.className, element);
+      const namespacedClassName = namespaceIdentifierName(className, node);
+      element.className = namespacedClassName;
+      // Set the element on both the namespaced & unnamespaced names so that we
+      // can detect registration by either name.
+      this._possibleElements.set(namespacedClassName, element);
+      this._possibleElements.set(className, element);
     }
   }
 
   enterClassDeclaration(node: estree.ClassDeclaration) {
     const element = this._handleClass(node);
     if (element) {
-      element.className = node.id.name;
-      this._possibleElements.set(element.className, element);
+      const className = node.id.name;
+      const namespacedClassName = namespaceIdentifierName(className, node);
+      element.className = namespacedClassName;
+      // Set the element on both the namespaced & unnamespaced names so that we
```

Comment:
What's the reasoning here?  That an element will be referenceable under both is namespaced and un-namespaced name? I think we want only one canonical identifier.